### DISCUSSION TITLE
gemini-2.0-flash-thinking-exp-1219 does not work

### DISCUSSION THREAD

**Discussion by andresgit** [danny-avila/LibreChat#5058] [Troubleshooting]
### What happened?

Trying gemini-2.0-flash-thinking-exp-1219 gives:
"Something went wrong. Here's the specific error message we encountered: An error occurred while processing your request. Please contact the Admin."

### Steps to Reproduce

1. try gemini-2.0-flash-thinking-exp-1219

### Screenshots

![image](https://github.com/user-attachments/assets/6f4a31d0-7208-480a-8770-7fe9e2ee9aa7)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
It works, but for some reason Google has made one of the safety settings a requirement.

![image](https://github.com/user-attachments/assets/3a248be9-f733-4229-8840-4c604ffb3668)

Besides adding the model to the list, you need to add the following to your `.env` file:

```.env
GOOGLE_SAFETY_CIVIC_INTEGRITY=BLOCK_NONE
```

Or whichever safety setting you prefer: 
https://ai.google.dev/gemini-api/docs/safety-settings

**Reply by danny-avila** [reply]
I'll make `BLOCK_NONE` the default value for this if not provided in the .env

**Reply by wsybok** [reply]
<img width="750" alt="Screenshot 2024-12-20 at 2 51 17 PM" src="https://github.com/user-attachments/assets/a0f21bbb-8bb8-47f5-9873-6c3f8c6ccdb5" />
Gemini's new "thinking-exp" model appears to generate a detailed thought process, while LibreChat functions as expected, producing standard output and counting as output tokens. I believe this difference is due to the use of the langchain package.  Perhaps we should consider migrating to the Google GenAI SDK?

**Reply by danny-avila** [reply]
This is expected.

it is actually using `@google/generative-ai` in this case
`const { GoogleGenerativeAI: GenAI } = require('@google/generative-ai');`
![Code_lzoykyZRJ1](https://github.com/user-attachments/assets/5777a056-bb61-4c34-9d26-62b34833ebab)

**Reply by danny-avila** [reply]
curl is the same

```bash
[
    {
        "candidates": [
            {
                "content": {
                    "parts": [
                        {
                            "text": "Here's a breakdown of my thought process in generating the Python list sorting code"
                        }
                    ],
                    "role": "model"
                },
                "index": 0,
                "safetyRatings": [
                    {
                        "category": "HARM_CATEGORY_SEXUALLY_EXPLICIT",
                        "probability": "NEGLIGIBLE"
                    },
                    {
                        "category": "HARM_CATEGORY_HATE_SPEECH",
                        "probability": "NEGLIGIBLE"
                    },
                    {
                        "category": "HARM_CATEGORY_HARASSMENT",
                        "probability": "NEGLIGIBLE"
                    },
                    {
                        "category": "HARM_CATEGORY_DANGEROUS_CONTENT",
                        "probability": "NEGLIGIBLE"
                    }
                ]
            }
        ],
        "usageMetadata": {
            "promptTokenCount": 9,
            "candidatesTokenCount": 16,
            "totalTokenCount": 25
        },
        "modelVersion": "gemini-2.0-flash-thinking-exp-1219"
    },
    {
        "candidates": [
            {
                "content": {
                    "parts": [
                        {
                            "text": " and explanation:\n\n1. **Understanding the Core Request:** The user wants to"
                        }
                    ],
                    "role": "model"
                },
```
